expr/random.go

```diff
@@ -4,25 +4,89 @@ import (
 	"crypto/md5"
 	"encoding/binary"
 	"math/rand"
+	"net"
+	"strings"
 
 	"github.com/manveru/faker"
 )
 
-// Random generates consistent random values of different types given a seed.
-// The random values are consistent in that given the same seed the same random values get
-// generated.
-// The generator tracks the user types that it has processed to avoid infinite recursions, this
-// means a new generator should be created when wanting to generate a new random value for a user
-// type.
-type Random struct {
-	Seed  string
-	Seen  map[string]*interface{}
-	faker *faker.Faker
-	rand  *rand.Rand
+// Randomizer generates consistent random values of different types given a seed.
+//
+// The random values should be consistent in that given the same seed the same
+// random values get generated.
+type Randomizer interface {
+	// ArrayLength decides how long an example array will be
```

Comment:
Looks like this isn't being actually used with the latest commit. The deterministic randomizer is probably broken with that last change.

Reply:
Oops. I got a bit desperate trying to figure out what part of my change had broken the tests and reverted too much of it. [9a5a6f9](https://github.com/goadesign/goa/pull/3191/commits/9a5a6f9cdfcba01a7a76ce2f5f0a510ad579f76d) brings back the use of ArrayLength, and I've tested this out on our design and it has the desired effect.